home/private_dot_local/lib/agent-equipment/agent_equipment/validator.py

```
@@ -0,0 +1,2533 @@
+            )
+            secret_value_expected = False
+            continue
+        return False
+    return (
+        not secret_value_expected
+        and len(consumed_secret_references) == len(set(consumed_secret_references))
+        and set(consumed_secret_references) == declared_secret_references
+        and (
+            command != "secret-exec"
+            or bool(consumed_secret_references)
+            and consumed_secret_references[0][0] == "secret_profile"
```

Ruff (0.16.1)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)